registry/handlers/manifests.go

```diff
@@ -481,12 +483,22 @@ func (imh *manifestHandler) DeleteManifest(w http.ResponseWriter, r *http.Reques
 		return
 	}
 
+	g := errgroup.Group{}
+	g.SetLimit(storage.DefaultConcurrencyLimit)
 	for _, tag := range referencedTags {
-		if err := tagService.Untag(imh, tag); err != nil {
-			imh.Errors = append(imh.Errors, err)
-			return
-		}
+		tag := tag
+
+		g.Go(func() error {
+			if err := tagService.Untag(imh, tag); err != nil {
+				imh.lock.Lock()
+				imh.Errors = append(imh.Errors, err)
+				imh.lock.Unlock()
+			}
+			return nil
+		})
 	}
+	// imh already records all errors, so ignore the error of Wait()
+	g.Wait() //nolint:errcheck
```

Comment:
Try to avoid using nolint annotations when more idiomatic constructs are available.
```suggestion
	// imh already records all errors, so ignore the error of Wait()
	_ = g.Wait()
```